Kernel/Devices/Device.h

```diff
@@ -25,6 +25,7 @@ class Device : public File {
     uid_t gid() const { return m_gid; }
 
     virtual bool is_device() const override { return true; }
+    bool is_disk_device() const { return false; }
```

Comment:
This must be `virtual` or the compiler will choose which `is_disk_device()` to call based on the current pointer type, not based on the underlying object type!